cmd/kind/load/image-archive/image-archive.go

```diff
@@ -96,17 +103,54 @@ func runE(flags *flagpole, cmd *cobra.Command, args []string) error {
 			selectedNodes = append(selectedNodes, node)
 		}
 	}
+	return loadImageOnNodes(imageTarPath, selectedNodes)
+}
+
+// loadImageOnNodes loads concurrently an image tarball in a group of nodes
+func loadImageOnNodes(imageTarName string, nodes []clusternodes.Node) error {
+	// Load the image into every node concurrently
+	// create all of the node containers, concurrently
+	errChan := make(chan error, len(nodes))
+	var wg sync.WaitGroup
+	for _, node := range nodes {
+		node := node // capture loop variable
+		// Don't load image on external nodes like the load balancer
+		nodeRole, err := node.Role()
+		if err != nil {
+			return err
+		}
+		if nodeRole == constants.ExternalLoadBalancerNodeRoleValue {
+			continue
+		}
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			// load the image on the node
+			err := loadImage(imageTarName, &node)
+			if err != nil {
+				errChan <- err
+				return
+			}
+			errChan <- nil
+		}()
+	}
+
+	// wait for all node creation to be done before closing
+	go func() {
+		defer close(errChan)
+		wg.Wait()
+	}()
 
-	// Load the image into every node
-	// TODO(bentheelder): this should probably be concurrent
-	for _, node := range selectedNodes {
-		if err := loadImage(args[0], &node); err != nil {
+	// return the first error encountered if any
```

Comment:
we've passed [WET](https://dev.to/wuz/stop-trying-to-be-so-dry-instead-write-everything-twice-wet-5g33), sending a PR to de-dupe this logic finally :-)

Reply:
https://github.com/kubernetes-sigs/kind/pull/481